index.js

```diff
@@ -48,6 +55,201 @@ class ExecutorQueue extends Executor {
+                        process.exit(128);
+                    }
+                    process.exit(0);
+                });
+            });
+        });
+    }
+
+    /**
+     * Posts a new build event to the API
+     * @method postBuildEvent
+     * @param {Object} config          Configuration
+     * @param {Object} config.pipeline Pipeline of the job
+     * @param {Object} config.job      Job object to create periodic builds for
+     * @return {Promise}
+     */
+    async postBuildEvent({ pipeline, job }) {
+        const jwt = this.tokenGen(pipeline.admins[0], {}, pipeline.scmContext);
+
+        const options = {
+            url: '/events',
+            method: 'POST',
+            headers: {
+                Authorization: `Bearer ${jwt}`,
+                'Content-Type': 'application/json'
+            },
+            body: {
+                pipelineId: pipeline.id,
+                startFrom: job.name
+            }
+        };
+
+        return req(options, (err, response) => {
+            if (!err && response.statusCode === 200) {
+                return Promise.resolve(response);
+            }
+
+            return Promise.reject(err);
+        });
+    }
+
+    /**
+     * Starts a new periodic build in an executor
+     * @method _startPeriodic
+     * @param {Object}   config             Configuration
+     * @param {Object}   config.pipeline    Pipeline of the job
+     * @param {Object}   config.job         Job object to create periodic builds for
+     * @param {Function} config.tokenGen    Function to generate JWT from username, scope and scmContext
+     * @param {Boolean}  config.isUpdate    Boolean to determine if updating existing periodic build
+     * @return {Promise}
+     */
+    async _startPeriodic(config, triggerBuild = false) {
+        // eslint-disable-next-line max-len
+        const buildCron = hoek.reach(config.job, 'permutations>0>annotations>screwdriver.cd/buildPeriodically',
+            { separator: '>' });
+
+        // Save tokenGen to current executor object so we can access it in postBuildEvent
+        if (!this.tokenGen) {
+            this.tokenGen = config.tokenGen;
+        }
+
+        if (config.isUpdate) {
+            // eslint-disable-next-line no-underscore-dangle
+            await this._stopPeriodic({
+                jobId: config.job.id
+            });
+            delete config.isUpdate;
+        }
+
+        if (triggerBuild) {
+            await this.postBuildEvent(config);
+        }
+
+        if (buildCron) {
+            await this.connect();
+
+            const next = cron.next(cron.transform(buildCron, config.job.id));
+
+            // Store the config in redis
+            await this.redisBreaker.runCommand('hset', this.periodicBuildTable,
+                config.job.id, JSON.stringify(config));
+
+            // Note: arguments to enqueueAt are [timestamp, queue name, job name, array of args]
+            return this.queueBreaker.runCommand('enqueueAt', next,
```

Comment:
From the implementation of [enqueueAt](https://github.com/taskrabbit/node-resque/blob/1b8509651bf5962b987cce2de819cd1d2c9576b7/lib/queue.js#L65-L66), it seems an exception is thrown if there is an attempt to enqueue the same item with the same timestamp. We may want to catch (and ignore if this is expected) here, so that the ones invoking this function (e.g. model.update()) will be able to proceed?

Reply:
Fixed, included a catch